relay-general/src/pii/processor.rs

```diff
@@ -195,6 +198,78 @@ impl<'a> Processor for PiiProcessor<'a> {
         replay.process_child_values(self, state)?;
         Ok(())
     }
+
+    fn process_event(
+        &mut self,
+        event: &mut Event,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        // TODO: make keys lazy init
+        let mut keys = Vec::new();
+
+        // collect the variables keys and scrub them out
+        if let Some(request) = event.request.value_mut() {
+            if let Some(Value::Object(data)) = request.data.value_mut() {
+                if let Some(Annotated(Some(Value::Bool(graph_ql_error)), _)) =
+                    request.other.get("graphql_error")
+                {
+                    dbg!("graphql_error request {graphql_error}");
+                    if *graph_ql_error {
+                        if let Some(Annotated(Some(Value::Object(variables)), _)) =
+                            data.get_mut("variables")
+                        {
+                            for item in variables.iter_mut() {
+                                keys.push(item.0.to_string());
+                                item.1
+                                    .set_value(Some(Value::String("[Filtered]".to_string())));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // scrub PII from the data object if they match the variables keys
+        if let Some(contexts) = event.contexts.value_mut() {
+            if let Some(Context::Response(response)) = contexts.get_context_mut("response") {
+                if let Some(Value::Object(data)) = response.data.value_mut() {
+                    if let Some(Annotated(Some(Value::Bool(graph_ql_error)), _)) =
+                        response.other.get("graphql_error")
+                    {
+                        if *graph_ql_error {
+                            if let Some(Annotated(Some(Value::Object(response_data)), _)) =
+                                data.get_mut("data")
+                            {
+                                scrub_data_from_response(keys.clone(), response_data);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        event.process_child_values(self, state)?;
+
+        Ok(())
+    }
+}
+
+fn scrub_data_from_response(keys: Vec<String>, data: &mut BTreeMap<String, Annotated<Value>>) {
+    for item in data.iter_mut() {
+        match item.1.value_mut() {
+            Some(Value::Object(item_data)) => {
+                // TODO: can we avoid cloning every time?
+                scrub_data_from_response(keys.clone(), item_data);
+            }
+            _ => {
+                if keys.contains(item.0) {
+                    item.1
+                        .set_value(Some(Value::String("[Filtered]".to_string())));
+                }
+            }
+        }
+    }
```

Comment:
Do we want these changes to be conditional on user configuration? If so, we'll have to think of how to pass the configuration through:

![image](https://github.com/getsentry/relay/assets/1565449/1cdb87e9-b3ca-410b-9c5a-09f50947f707)

Reply:
Since we are guarding against `graphql_error`, I believe not for now, we can add it later if needed.